### PR TITLE
feat(genesis bridge): genesis transfers

### PR DESCRIPTION
**Pull request by danwt**
## Description

See

https://github.com/dymensionxyz/research/issues/278 for the master issue.

There are a few outstanding items, that this PR does not try to solve.

This PR:

- do not allow opening any ibc connections starting at the RA side
- hard codes alice/bob keys in test scripts to make the address deterministic
- adds `-y` to register rollapp / register sequencer cmds
- wires in the new hub genesis ibc module (see RDK pr https://github.com/dymensionxyz/dymension-rdk/pull/449)

NOTE: I will update the go.mod to the correct dependencies after the other prs are merged

----

Closes https://github.com/dymensionxyz/rollapp-evm/issues/278

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)
- [ ]  Updated the scripts for local run, e.g genesis_config_commands.sh if the PR changes parameters
- [ ]  Add an issue in the [e2e-tests repo](https://github.com/dymensionxyz/e2e-tests) if necessary

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys
- [ ] UInt to byte conversion should use BigEndian

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

----;

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

---;

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

## Summary by CodeRabbit

- **New Features**
  - Introduced a message type check to prevent connection openings from one side.

- **Improvements**
  - Updated scripts for setting up IBC paths and registering rollapps, sequencers, and genesis accounts with new flags for automation and better organization.
  - Enhanced genesis file updates for account balances, metadata, and elevated addresses.

- **Permissions**
  - Modified permissions for specific module names.
  - Adjusted `TransferKeeper` initialization for better control.

### CODE REVIEW

File: scripts/trigger_rollapp_genesis_event.sh
Comment (srene) [inline scripts/trigger_rollapp_genesis_event.sh:1]: i think we need to update the readme file to be aligned with the changes. also if you could update the block time params ...

File: go.mod
Comment (danwt) [reply go.mod:334]: I will after the others prs are merged

File: scripts/trigger_rollapp_genesis_event.sh
Comment (danwt) [reply scripts/trigger_rollapp_genesis_event.sh:1]: will do

**Comment by danwt** [comment]
Thanks @srene I've fixed the readme now

File: README.md
Comment (mtsitrin) [inline README.md:148]: not sure we need to introduce additional dependency here
i'd keep the `sed` cmd

**Review by mtsitrin — DISMISSED** [review]

File: scripts/ibc/setup_ibc.sh
Comment (danwt) [inline scripts/ibc/setup_ibc.sh:99]: @zale144

File: README.md
Comment (zale144) [inline README.md:145]: will this work on mac?

File: scripts/settlement/register_rollapp_to_hub.sh
Comment (zale144) [inline scripts/settlement/register_rollapp_to_hub.sh:24]: shouldn't this also have `--transfers-enabled true` ?

File: README.md
Comment (danwt) [reply README.md:145]: sorry yes meant to delete the linux word

File: scripts/settlement/register_rollapp_to_hub.sh
Comment (danwt) [reply scripts/settlement/register_rollapp_to_hub.sh:24]: no
but now it's different anyway after
https://github.com/dymensionxyz/dymension/pull/961
https://github.com/dymensionxyz/dymension-rdk/pull/458

File: app/ante/handlers.go
Comment (danwt) [reply app/ante/handlers.go:47]: I think ante is fine
If you think that's better I will think, anyways would prefer to change it in another PR

File: README.md
Comment (danwt) [reply README.md:210]: sure sure, out of scope

File: README.md
Comment (danwt) [reply README.md:169]: 🤷

File: app/app.go
Comment (zale144) [inline app/app.go:550]: can we pls use the original signature instead of wrapping

File: scripts/update_genesis_file.sh
Comment (zale144) [inline scripts/update_genesis_file.sh:29]: I don't think we have `denom_metadata.params...` anymore

File: app/app.go
Comment (danwt) [reply app/app.go:549]: sure I will change

File: scripts/update_genesis_file.sh
Comment (danwt) [reply scripts/update_genesis_file.sh:29]: do we or don't we?

Comment (danwt) [reply scripts/update_genesis_file.sh:29]: I'll leave it to you to finish this please @zale144 
As it corresponds to your denom meta data work

**Review by mtsitrin — APPROVED** [review]

File: scripts/update_genesis_file.sh
Comment (danwt) [inline scripts/update_genesis_file.sh:23]: @danwt

**Pull request merged**
